app/src/main/java/com/example/practiceapp/LoginActivity.kt

```diff
@@ -19,47 +19,42 @@ class LoginActivity : AppCompatActivity() {
+            startActivity(Intent(this, ActivityRegister::class.java))
         }
 
         binding.loginBtnLogin.setOnClickListener {
-
             val emaill = binding.loginEtUser.text.toString()
-            val pass = binding.loginEtpass.text.toString()
-            val db = (application as UserApp).database
-            val userdb = db.userDao().getById(emaill)
-            val emaildb = userdb.email
-            val passdb = userdb.password
-            if(emaildb.isBlank()){
-                if (emaill == emaildb && passdb == pass){
-                    startActivity(Intent(this, BuscarActivity::class.java))
+            val pass = binding.loginEtPass.text.toString()
+            if (emaill.isEmpty() || pass.isEmpty()) {
+                Toast.makeText(this, "Llenar todos los campos", Toast.LENGTH_SHORT).show()
+            } else {
+                val db = (application as UserApp).database
+                val userDb = db.userDao().getByIDAnDPass(emaill,pass)
+                if (userDb != null){
+                    startActivity(Intent(this, SearchActivity::class.java))
                 }else{
-                    Toast.makeText(this, " Usuario o contraseña incorrecto", Toast.LENGTH_SHORT).show()
+                    Toast.makeText(this, "Usuario o contraseña incorrecta", Toast.LENGTH_SHORT).show()
                 }
-            }else{
-                Toast.makeText(this,"usuario o contraseña incorrectos", Toast.LENGTH_SHORT).show()
-            }
-
-
 
+            }
         }
         binding.loginEtUser.addTextChangedListener(object : TextWatcher{
 
             override fun afterTextChanged(s: Editable?) {}
             override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
             override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
-                if (binding.loginEtUser.text?.length!! > 0 && binding.loginEtpass.text?.length!! > 0 ){
+                if (binding.loginEtUser.text?.length!! > 0 && binding.loginEtPass.text?.length!! > 0 ){
                     binding.loginBtnLogin.setBackgroundColor(Color.parseColor("#4B4747"))
                 }else{
                     binding.loginBtnLogin.setBackgroundColor(Color.parseColor("#9D9C9C"))
                 }
             }
         })
-        binding.loginEtpass.addTextChangedListener(object : TextWatcher{
+        binding.loginEtPass.addTextChangedListener(object : TextWatcher{
             override fun afterTextChanged(s: Editable?) {}
             override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
             override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
-                if (binding.loginEtUser.text?.length!! > 0 && binding.loginEtpass.text?.length!! > 0 ){
+                if (binding.loginEtUser.text?.length!! > 0 && binding.loginEtPass.text?.length!! > 0 ){
```

Comment:
Lines 46 and 57 have the same logic, you can move it to a function and call it from both places to avoid repeating code

Reply:
and same for `RegisterActivity`, now as this validation is required in two different classes you can have a Utils class with this validation as a function and then calling it from different places